src/lib.rs

```diff
@@ -1390,6 +1392,85 @@ impl Default for StashFlags {
     }
 }
 
+macro_rules! define_attr_inspector {
+    ($enum_name:ident, $enum_doc:literal, $fn_name:ident, $fn_doc:literal, $s:ty) => {
+        #[derive(Debug, Clone, Copy, PartialEq, Eq)]
+        #[doc = $enum_doc]
+        pub enum $enum_name<'string> {
+            /// The attribute is set to true.
+            True,
+            /// The attribute is unset (set to false).
+            False,
+            /// The attribute is set to a string.
+            String(&'string $s),
+            /// The attribute is not specified.
+            Unspecified,
+        }
+
+        #[doc = $fn_doc]
+        pub fn $fn_name(attr_value: Option<&$s>) -> $enum_name<'_> {
```

Comment:
Perhaps this should be an inherent method on `$enum_name`, for example `AttrValue::new(Some("bar"))`?